components/ManageAvailabilityWizard/AvailabilityModal.tsx

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable import/extensions */
+import React, { useState, useEffect, useContext } from "react";
+import styles from "./AvailabilityModal.module.css";
+
+import "react-date-picker/dist/DatePicker.css";
+import "react-calendar/dist/Calendar.css";
+import ClipLoader from "react-spinners/ClipLoader";
+import { DayRow } from "./DayRow";
+import { DateTime, StringUnitLength} from "luxon";
+import useSWR from "swr";
+import { APIContext } from "../../context/APIContext";
+import { Loader } from "../util/Loader";
+import { Error } from "../util/Error";
+import { AuthContext } from "../../context/AuthContext";
+
+type Availability = {
+  [Day: string]: string[][];
+  
+}
+
+
+type AvailabilityModalProps = {
+  handleClose: () => void;
+  // handleStates: (availability: Availability, timeZone: string) => void;
+  // props for initial state values
+  initAvailability: Availability;
+  initTimeZone: string;
+  userId: string;
+  mutate: (availabiliy:any) => void;
+};
+
+const AvailabilityModal: React.FC<AvailabilityModalProps> = ({
+  handleClose,
+  // handleStates,
+  initAvailability,
+  initTimeZone,
+  userId,
+  mutate
+}) => {
+  // repeat modal states initialized by init props
+  const [availability, setAvailability] = useState(initAvailability);
+  const [timezone, setTimeZone] = useState(initTimeZone);
+  const [loading, setLoading] = useState(false)
+  const [valid, setValid] = useState(true)
+  const [errMsg, setErrMsg] = useState("")
+
+  const client = useContext(APIContext);
+
+  const sortTimes = (times: string[][]): string[][] => {
+    times.sort((t1:string[], t2:string[])=>{
+      if (t1[0] == t2[0]){
+        return 0;
+      }
+      return t1[0] < t2[0] ? -1 : 1;
+    })
+
+    return times
+  }
+  
+  const areTimesValid = (availability: Availability): [boolean, string, string] => {
```

Comment:
can we have a comment here specifying what the return values mean. Like what the boolean, string, and string actually specify